test/conformance/serial/ptp.go

```diff
@@ -1345,23 +1348,116 @@ var _ = Describe("["+strings.ToLower(DesiredMode.String())+"-serial]", Serial, f
 					Skip("test valid only for GM test config")
 				}
 			})
-			PIt("Verify Events during GNSS Loss flow (V1)", func() {
-
-			})
-
 		})
 
 		Context("WPC GM Events verification (V2)", func() {
 			BeforeEach(func() {
 				if fullConfig.PtpModeDiscovered != testconfig.TelcoGrandMasterClock {
 					Skip("test valid only for GM test config")
 				}
+
+				// Set up consumer pod for event monitoring
+				if fullConfig.DiscoveredClockUnderTestPod != nil {
+					nodeName := fullConfig.DiscoveredClockUnderTestPod.Spec.NodeName
+					if nodeName != "" {
+						logrus.Info("Deploy consumer app for testing event API v2")
+						err := event.CreateConsumerApp(nodeName)
+						if err != nil {
+							logrus.Errorf("PTP events are not available due to consumer app creation error err=%s", err)
+							Skip("Consumer app setup failed")
+						}
+
+						// Wait a bit more for the consumer pod to be fully ready
+						logrus.Info("Waiting for consumer pod to be fully ready...")
+						time.Sleep(10 * time.Second)
+
+						// Initialize pub/sub system
+						event.InitPubSub()
+					}
+				}
 			})
-			PIt("Verify Events during GNSS Loss flow (V2)", func() {
 
+			AfterEach(func() {
+				// Clean up consumer namespace
+				DeferCleanup(func() {
+					err := event.DeleteConsumerNamespace()
+					if err != nil {
+						logrus.Debugf("Deleting consumer namespace failed because of err=%s", err)
+					}
+				})
+
+				// Close internal pubsub
+				if event.PubSub != nil {
+					event.PubSub.Close()
+				}
 			})
-			PIt("Verify Events during GNSS holdover state  (V2)", func() {
 
+			It("Verify Individual Events (Clock Class, GNSS, PTP State)", func() {
+				By("Testing individual event verification for Clock Class, GNSS, and PTP State")
```

Comment:
Use By for each step, like:
check locked:
start cold boot
listen for events until timeout
stop cold boot
listen for events until timeout